app/controllers/user_registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+class UserRegistrationsController < ApplicationController
+  def edit
+    @token = params[:id]
+    @user = User.find_by(registration_token: params[:id])
+
+    if @user.blank?
+      not_authenticated
+      return
+    end
+
+  end
+
+  def update
+
+    @token = params[:id]
+    @user = User.find_by(registration_token: params[:id])
+
+    if @user.blank?
+      not_authenticated
+      return
+    end
+    @user.registration_token = nil
```

Comment:
instead of setting registration_token and password_confirmation here you can use `tap` in user_params
example:
```ruby
def product_params
  params.require(:product).permit(:name).tap do |whitelisted|
    whitelisted[:data] = params[:product][:data]
  end
end
```